packages/stitch/src/mergeCandidates.ts

```diff
@@ -84,6 +91,17 @@ function mergeObjectTypeCandidates(
   const interfaces = Object.keys(interfaceMap).map(interfaceName => interfaceMap[interfaceName]);
 
   const astNodes = pluck<ObjectTypeDefinitionNode>('astNode', candidates);
+  const fieldAstNodes = Object.values(fields)
+    .map(({ astNode }) => astNode)
+    .filter(n => n != null);
+
+  if (astNodes.length > 1 && fieldAstNodes.length) {
+    astNodes.push({
+      ...astNodes[astNodes.length - 1],
+      fields: JSON.parse(JSON.stringify(fieldAstNodes)),
```

Comment:
Do we need to recreate? Are we modifying these?

Reply:
the reason this is done is because logic now selects all the final field definition values that we want (directive arg values, field nullability, etc). We still need to go through the merge process so that all fields have their characteristics combined, and that is an incrementally destructive process on the original AST nodes. So, before merging this clean copy is made and added as the final merge step... All fields will be merged as normal, and then the original selection of preferred values will be applied at the end.

Reply:
I didn't follow this. So I just tried removing it, and sure enough, your (comprehensive!) test suite fails.

It seems like the mergeType (et al) functions did not quite do what I expected, i.e. modify the passed in fields, not just in the target of the merge, but also the sources?

And this is a work-around for that?

@ardatan is this the work-around you would suggest?

Reply:
Yes, this is a work-around for the fact that merge methods modify the original AST objects. By creating an AST copy of those fields up front, we can reapply them at the end.

I just added an update that makes this way more efficient... Now [subschema config is checked for canonical field overrides](https://github.com/ardatan/graphql-tools/pull/2417/files#diff-0c6208cac22fe2f4399a4f1a23bdf213b4162d8985fcf63e750158e9aa16a1f6R93) (which should be fairly uncommon), and only those specific field overrides are reapplied via this copy patch. So, it's a lot more efficient now than always reapplying a copy of all fields.

Reply:
I am ready to merge this, it looks great, my last concern is to wait for @ardatan if he has time to take a look, if these merge functions are "supposed" to work like this, and if there should be another fix. Or we could create a separate issue to track like for mergeEnums

Reply:
going to merge for now, we might need to revisit if @ardatan has a chance to weigh in, but i think that will be just internal implementation details.

long-term, we might want to actually figure out a way to remove the `@stitch` package's dependency on `@merge` anyway...